docs/preview/02-Features/06-general-messaging.md

```diff
@@ -78,6 +78,31 @@ Both the recovery period after the circuit is open and the interval between mess
 }
 ```
 
+#### 🔔 Get notified on a circuit breaker state transition
+Transitions from circuit breaker states happens internally and automatically. The library supports a notification system that lets you register event handlers that gets called upon a state transition.
+
+After the message pump and/or message handlers are registered, you can add one or more handlers linked to specific transitions to the previously registered pump.
+
+```csharp
+using Arcus.Messaging.Pumps.Abstractions;
+
+services.AddServiceBusMessagePump(...)
+        .WithCircuitBreakerEventHandler<MyFirstCircuitBreakerEventHandler>()
```

Comment:
Should we specify what kind of event-handler this is ?  Or make it clear that this is an event-handler that is triggered when the CircuitBreakerState has changed ?

`WithCircuitBreakerStateChangedEventHandler` or something ?